packages/api-bindings/src/graphql/utils/publication.ts

```diff
@@ -93,8 +98,156 @@ export type PublicationOwnedByMeFragment = Overwrite<
   { profile: Overwrite<ProfileFragment, { ownedByMe: true }> }
 >;
 
-export const isPublicationOwnedByMe = (
+export function isPublicationOwnedByMe(
   publication: PublicationFragment,
-): publication is PublicationOwnedByMeFragment => {
+): publication is PublicationOwnedByMeFragment {
   return publication.profile.ownedByMe;
+}
+
+export function createCollectRequest(
+  publication: PublicationFragment,
+  activeProfile: ProfileFragment,
```

Comment:
"Active profile" is a concept of the state management done by the `react` package. One can collect with any profile they own as far as they can sign the tx request (also not event signing required if they use a proxied collect action).

`collector` maybe?

Reply:
nice idea 👍 added